### PR TITLE
XJS -> JSX

### LINKED ISSUE #83: Move from XJS prefixes to JSX
Following talk with @sebmarkbage, decided to bring this issue back.

Earlier the only thing I was worried about was that this might break backward compatibility for other AST tools, but now [Espree](https://github.com/eslint/espree) (<=> [ESLint](https://github.com/eslint/eslint)) has already [switched from XJS to JSX prefixes](https://github.com/eslint/eslint/commit/f90d1537a50df224137792f2bd6dcf3e781dfffa#commitcomment-9262570) for internal needs.

In [Acorn-JSX](https://github.com/RReverser/acorn-jsx) (<=> [6to5](https://github.com/6to5/6to5)) I'd like to do the same, but wouldn't want to do that without agreement with primary player (Esprima-FB), as it exposes AST to outer consumers.

### PR DESCRIPTION
**Pull request by zpao**
Simple find+replace across the codebase. Tests pass. We'll want to bump major versions for this, but otherwise it seems pretty ok.

Fixes #83

cc @jeffmo @spicyj @sebmarkbage @gabelevi

### CODE REVIEW

File: test/fbtest.rec.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
 * WARNING: This file is autogenerated by tools/generate-fbtest.js
-* Do NOT modify this file directly! Instead, add your tests to 
+* Do NOT modify this file directly! Instead, add your tests to
```
Comment (sophiebits) [inline test/fbtest.rec.js:3]: dun dun dun

**Comment by zpao** [comment]
See https://github.com/facebook/esprima/pull/86#issuecomment-71270117 for why this is failing in node 0.11

**Comment by jeffmo** [comment]
woohoo!

**Comment by jeffmo** [comment]
What changed in `test/3rdparty/threejs-r51.js`?

**Comment by RReverser** [comment]
I guess it's also worth to mention @benjamn so he could update [ast-types defs](https://github.com/benjamn/ast-types/blob/master/def/fb-harmony.js) as well.
EDIT: Ah, just figured out from his profile that he doesn't work for FB anymore.

**Comment by RReverser** [comment]
@zpao @jeffmo Are there any issues with this?

**Comment by jeffmo** [comment]
The only issue right now is that it's failing the Travis build

**Comment by zpao** [comment]
Which is a cascading issue - we need to fix #87 so we can merge upstream in again more cleanly, which updates the linting process, which makes the travis build pass.

**Comment by zpao** [comment]
Rebased and tests passing. We'll want to get our JSX transform in React changed (which is annoying with dependencies and having it in multiple places...). But otherwise, I think we can this in any time.

**Pull request merged**